src/app/shared/components/List/index.tsx

```diff
@@ -164,26 +164,50 @@ ListProps<T, F>): JSX.Element {
     'ctrl+a, cmd+a',
     (e) => {
       e.preventDefault();
-      selectAllItems();
+      if (!disableKeyboardShortcuts) selectAllItems();
     },
-    [items, selectedItems],
+    [items, selectedItems, disableKeyboardShortcuts],
   );
 
-  useHotkeys('esc', unselectAllItems, [selectedItems]);
+  useHotkeys(
+    'esc',
+    () => {
+      if (!disableKeyboardShortcuts) unselectAllItems();
+    },
+    [selectedItems, disableKeyboardShortcuts],
+  );
 
-  useHotkeys('enter', executeClickOnSelectedItem, [selectedItems]);
+  useHotkeys(
+    'enter',
+    () => {
+      if (!disableKeyboardShortcuts) executeClickOnSelectedItem();
+    },
+    [selectedItems, disableKeyboardShortcuts],
+  );
 
   const handleRKeyPressed = () => {
     keyBoardShortcutActions?.onRKeyPressed?.();
   };
 
-  useHotkeys('r', handleRKeyPressed, [selectedItems]);
+  useHotkeys(
+    'r',
+    () => {
+      if (!disableKeyboardShortcuts) handleRKeyPressed();
+    },
+    [selectedItems, disableKeyboardShortcuts],
+  );
 
   const handleBackspaceKeyPressed = () => {
     keyBoardShortcutActions?.onBackspaceKeyPressed?.();
   };
 
-  useHotkeys('backspace', handleBackspaceKeyPressed, [selectedItems]);
+  useHotkeys(
+    'backspace',
+    () => {
+      if (!disableKeyboardShortcuts) handleBackspaceKeyPressed();
+    },
+    [selectedItems, disableKeyboardShortcuts],
+  );
```

Comment:
Maybe we could simplify the use of `!disabledKeyboardShortcuts` conditional in this way:
```
const handleKeyPress = (shortcutKey, action) => {
  return () => {
    if (!disableKeyboardShortcuts) action();
  };
};

useHotkeys('esc', handleKeyPress('esc', unselectAllItems), [selectedItems, disableKeyboardShortcuts]);
useHotkeys('enter', handleKeyPress('enter', executeClickOnSelectedItem), [selectedItems, disableKeyboardShortcuts]);
useHotkeys('r', handleKeyPress('r', handleRKeyPressed), [selectedItems, disableKeyboardShortcuts]);
useHotkeys('backspace', handleKeyPress('backspace', handleBackspaceKeyPressed), [selectedItems, disableKeyboardShortcuts]);
```